code/game/gamemodes/vampire/vampire_powers.dm

```diff
@@ -827,6 +837,31 @@
 			to_chat(src, "<span class='warning'>You are no longer influencing those weak of mind.</span>")
 			break
 
+/mob/living/carbon/human/proc/vampire_touch_of_life()
+	set category = "Vampire"
+	set name = "Touch of Life (50)"
+	set desc = "You lay your hands on the target, transferring healing chemicals to them."
+
+	var/datum/vampire/vampire = vampire_power(50, 0)
```

Comment:
This here just checks if the vampire has sufficient blood but is not actually using it.
You need to add a `vampire.use_blood(50)` when you actually perform the action.